packages/mdc-dialog/foundation.js

```diff
@@ -82,6 +82,7 @@ export default class MDCDialogFoundation extends MDCFoundation {
   }
 
   open() {
+    this.isOpen_ = true;
```

Comment:
That's a good idea. My thinking was that the dialog wouldn't open until all of the logic had been run successfully. I think this is a good approach though.